### ISSUE TITLE
clang 18: Generated C++ spaceship operator triggers clang-tidy warning

### ISSUE THREAD

**Issue by pkl97** [llvm/llvm-project#93409]
This program triggers a readability-implicit-bool-conversion clang-tidy warning:
```c++
#include <compare>
#include <list>

struct X
{
    auto operator<=>(const X&) const = default;
    bool m_b;
};

struct Y
{
    auto operator<=>(const Y&) const = default;
    std::list<X> m_xlist;
};

int main()
{
    return 0;
}
```

The warning emitted is the following:
```bash
main.cpp:6:46: warning: implicit conversion 'bool' -> 'int' [readability-implicit-bool-conversion]
    6 |     auto operator<=>(const X&) const = default;
      |                                              ^
      |                                              static_cast<int>( )
```

The clang-tidy command used:
```bash
clang-tidy --checks=-*,readability-implicit-bool-conversion main.cpp
```

It is somewhat unfortunate that a generated function triggers such a warning. Of course one can add NOLINT() statements to all affected operators but that unnecessarily clutters the code and makes it harder to read.

Is it possible to alter the generated code and avoid this warning?

**Comment by PiotrZSL** [comment]
No, check need to be fixed.

**Comment by pkl97** [comment]
Of course, this would also be okay.

I just thought it would be better to fix the root cause and not the resulting symptom.
But the readability aim of the check is questionable here anyway because the generated spaceship operator is not visible to the developer.
